### ISSUE TITLE
Using the same SQS queue for multiple CodeCommit repositories

### ISSUE THREAD

**Issue by ronaldtse** [riboseinc/aws-sqs-plugin#13]
Currently the plugin requires 1 SQS queue for 1 CodeCommit repo. It would be much more efficient if we only use 1 SQS queue for N CodeCommit repos.

This will require adding to the Job page, a textbox after the SQS queue selector but before the branch textbox, used to match the CodeCommit repo. The default CodeCommit repo selected should be the one used in the Git plugin for the job. Perhaps there is a way to determine which CodeCommit repos publish to a selected SQS queue.

**Comment by phuonghuynh** [comment]
@ronaldtse the original plugin works for all type of Jenkins Project, including "FreeStyleProject" (which no need to have SCM config)

The reason for this issue is that the original Plugin delete SQS messages after handling them.
see https://github.com/jenkinsci/aws-sqs-plugin/blob/master/src/main/java/io/relution/jenkins/awssqs/threading/SQSQueueMonitorImpl.java#L180

I think a good solution is that we should detect if JenkinsProject has SCM config, then we match its branch as well as SubscribeBranches

**Comment by ronaldtse** [comment]
I agree @phuonghuynh , maybe this plugin should be called `aws-codecommit-trigger-plugin` instead?

**Comment by phuonghuynh** [comment]
I think thats a better name, but we could publish new one?

**Comment by ronaldtse** [comment]
I think the plugin is quite different enough to be a new one... We're not sure if our requests will get merged, there has been no feedback.

**Comment by ronaldtse** [comment]
@phuonghuynh just made a new repo. Could you help migrate things over?

**Comment by phuonghuynh** [comment]
Yes, i can help to migrate the code. Could you help to update README to let ppl know that we are using the current plugin to create our new one?

**Comment by ronaldtse** [comment]
Sure, I'll update the README.

**Comment by ronaldtse** [comment]
Could you help provide a way of packaging / releasing the plugin? We could potentially serve the hpi file in Releases?

**Comment by phuonghuynh** [comment]
Yes, i will update README for it

**Comment by phuonghuynh** [comment]
Ping me when you update README file, then i will start migrating the code
> Sure, I'll update the README.

**Comment by phuonghuynh** [comment]
We can submit our plugin to Jenkins team for release
> We could potentially serve the hpi file in Releases?
